course-mw/course-actors-common/src/main/java/org/sunbird/learner/actors/coursebatch/dao/impl/BatchUserDaoImpl.java

```diff
@@ -0,0 +1,60 @@
+package org.sunbird.learner.actors.coursebatch.dao.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.collections.CollectionUtils;
+import org.sunbird.cassandra.CassandraOperation;
+import org.sunbird.common.CassandraUtil;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.request.RequestContext;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.learner.actors.coursebatch.dao.BatchUserDao;
+import org.sunbird.learner.util.Util;
+import org.sunbird.models.batch.user.BatchUser;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BatchUserDaoImpl implements BatchUserDao {
+
+
+    private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+    private Util.DbInfo batchUserDb = Util.dbInfoMap.get(JsonKey.BATCH_USER_DB);
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public BatchUser read(RequestContext requestContext, String batchId, List<String> userId) {
+
+        Map<String, Object> primaryKey = new HashMap<>();
+        primaryKey.put(JsonKey.BATCH_ID, batchId);
+        Response response = cassandraOperation.getRecordByIdentifier(requestContext, batchUserDb.getKeySpace(), batchUserDb.getTableName(), primaryKey, null);
+        List<Map<String, Object>> batchUserList =
```

Comment:
Please add logs and comments in the method

Reply:
done